src/ts/components/timelinemarkershandler.ts

```diff
@@ -121,20 +121,30 @@ export class TimelineMarkersHandler {
         .map(cssClass => this.prefixCss(cssClass));
 
       const cssProperties: { [propertyName: string]: string } = {
-        'width': marker.position + '%',
+        'left': `${marker.position}%`,
       };
 
       if (marker.duration > 0) {
         const markerWidthPx = Math.round(seekBarWidthPx / 100 * marker.duration);
-        cssProperties['border-right-width'] = markerWidthPx + 'px';
-        cssProperties['margin-left'] = '0';
+        cssProperties['width'] = `${markerWidthPx}px`;
       }
 
-      this.markersContainer.append(new DOM('div', {
+      const markerElement = new DOM('div', {
         'class': markerClasses.join(' '),
         'data-marker-time': String(marker.marker.time),
         'data-marker-title': String(marker.marker.title),
-      }).css(cssProperties));
+      }).css(cssProperties);
+
+      if (marker.marker.imageUrl) {
+        const imageElement = new DOM('img', {
+          'class': this.prefixCss('seekbar-marker-image'),
+          'src': marker.marker.imageUrl,
```

Comment:
One more thing maybe as a discussion point: Do we want to have any error handling in case the `imageUrl` is invalid or there is an error on `onLoad`?

Reply:
I guess in that case, we should not display any image

Reply:
Hmm, I can remove the image `onerror`

Reply:
Done in https://github.com/bitmovin/bitmovin-player-ui/pull/318/commits/8cd365f7f3c2f8846d4ed09af50a0b0c2e2a772d